### DISCUSSION TITLE
Compiling Babel to WebAssembly, then native.

### DISCUSSION THREAD

**Discussion by trusktr** [babel/babel#14080] [Ideas]
If Babel moves to a completely type-safe TypeScript code base, it will become possible to compile it to WebAssembly with [AssemblyScript](https://assemblyscript.org/), and with some additional steps to native eventually.

This could unlock performance enhancements.

Wdyt?

**Comment by lxmarinkovic** [comment]
@trusktr take a look at this one https://github.com/swc-project/swc :D
